app/presenters/flow_presenter.rb

```diff
@@ -23,7 +23,12 @@ def finished?
   end
 
   def current_state
-    @current_state ||= @flow.process(all_responses)
+    @current_state ||= if use_session?
+                         requested_node = params[:node_name] unless params[:next]
+                         @flow.resolve_state(params[:responses], requested_node)
+                       else
+                         @flow.process(all_responses)
+                       end
```

Comment:
Instead of an `if...else` here (and again in `change_collapsed_question_link`) would sub-classing the `FlowPresenter` and overriding these two methods be worthwhile?

Reply:
That's an interesting idea. However, I think that should be done as a separate PR where it's impact can be considered in isolation.